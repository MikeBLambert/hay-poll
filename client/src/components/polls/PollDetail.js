import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';

export default class PollDetail extends PureComponent {
  static propTypes = {
    poll: PropTypes.object.isRequired,
    submitVote: PropTypes.func.isRequired
  };

  state = {
    choices: [],
    buttonType: 'radio'
  };

  handleSubmit = event => {
    const { choices } = this.state;
    event.preventDefault();
    this.props.submitVote(this.props.match.params.id, choices);
  };

  handleInputChange = ({ target }) => {
    let newChoices = this.state.choices.concat(target.value);
    this.setState({ choices: newChoices });
  };

  render() {
    const { poll } = this.props;
    const { title, options } = poll;

    const optionList = options.map((option, i) => {
      return (
        <Fragment key={i}>
          <label>{option}</label>
          <input
            type={this.state.buttonType}
            value={option}
            onChange={this.handleInputChange}
          />
        </Fragment>
      );
    });

    return (
      <div>
        <h1>{title}</h1>
        <form onSubmit={this.handleSubmit}>
          <ul>{optionList}</ul>
          <button>Vote!</button>
        </form>
      </div>
    );
  }
}
