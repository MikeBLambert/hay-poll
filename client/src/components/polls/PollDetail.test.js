import React from 'react';
import PollDetail from './PollDetail';
import { shallow } from 'enzyme';

describe('Poll component', () => {
  it('renders a Poll component', () => {
    const poll = {
      title: 'Best Show of All Time',
      options: ['GOT', 'Seinfeld', 'The Simpsons']
    };
    const submitVote = jest.fn();

    const wrapper = shallow(<PollDetail poll={poll} submitVote={submitVote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
