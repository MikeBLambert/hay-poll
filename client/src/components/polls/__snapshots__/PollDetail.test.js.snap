// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll component renders a Poll component 1`] = `
<div>
  <h1>
    Best Show of All Time
  </h1>
  <form
    onSubmit={[Function]}
  >
    <ul>
      <label>
        GOT
      </label>
      <input
        onChange={[Function]}
        type="radio"
        value="GOT"
      />
      <label>
        Seinfeld
      </label>
      <input
        onChange={[Function]}
        type="radio"
        value="Seinfeld"
      />
      <label>
        The Simpsons
      </label>
      <input
        onChange={[Function]}
        type="radio"
        value="The Simpsons"
      />
    </ul>
    <button>
      Vote!
    </button>
  </form>
</div>
`;
